src/halo-hooks/useRewards.ts

```diff
@@ -9,53 +9,47 @@ import { ZERO_ADDRESS } from '../constants'
 import { BigNumber } from 'ethers'
 import { PENDING_REWARD_FAILED } from 'constants/pools'
 
-/**
- * Internal Methods
- */
-
-const usePoolLength = () => {
-  const rewardsContract = useHALORewardsContract()
-  const data = useSingleCallResult(rewardsContract, 'poolLength')
-
-  return useMemo<number>(() => {
-    return data.result ? data.result[0].toNumber() : 0
-  }, [data])
-}
-
-const useLpToken = (poolLength: number): string[] => {
+export const useLPTokenAddresses = () => {
   const rewardsContract = useHALORewardsContract()
-
-  const args = useMemo(() => {
-    const pids: string[][] = []
-    for (let i = 0; i < poolLength; i++) {
-      pids.push([`${i}`])
+  const [poolLength, setPoolLength] = useState(0)
+  const [lpTokenAddresses, setLpTokenAddresses] = useState<string[]>([])
+
+  const fetchPoolLength = useCallback(async () => {
+    try {
+      const result = await rewardsContract?.poolLength()
+      setPoolLength(result ? result.toNumber() : 0)
+    } catch (err) {
+      console.error(`Error fetching poolLength: `, err)
```

Comment:
I think this should be

```suggestion
      console.error(`Error fetching poolLength: ${err}`)
```

Reply:
@bitcoinbrisbane I would prefer this to be a parameter so that it prints nicely on the browser console log & show object tree, instead of a json string representation of the object